documentation/coding_guidelines.md

```diff
@@ -207,7 +207,9 @@ Generics allow dynamic behavior (similar to [`trait`](https://doc.rust-lang.org/
 
 Excluding test code, set field visibility to private as much as possible. Private fields allow constructors to enforce internal invariants. Implement getters for data that consumers may need, but avoid setters unless a mutable state is necessary.
 
-Public fields are most appropriate for [`struct`](https://doc.rust-lang.org/book/ch05-00-structs.html) types in the C spirit: compound, passive data structures without internal invariants.  Naming suggestions follow the guidance [here](https://rust-lang-nursery.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter) as shown below.
+Getters/setters should be avoided for [`struct`](https://doc.rust-lang.org/book/ch05-00-structs.html) types in the C spirit: compound, passive data structures without internal invariants. Adding them only increases the complexity and number of lines of code without improving the developer experience.
```

Comment:
I sort of just want to cut this paragraph, I don't think it ads much value imo

Reply:
Unfortunately I see a lot of folks writing get_ functions -- maybe 897 in the code base :O

Reply:
maybe explain that (non-)mutability should be enforced via the absence or presence of the `mut` keyword.

Reply:
You mean on the data structure itself? I'll think about how to shove that in, but I can see this being the reason for even having this blurb.

Reply:
on a variable referring to that data structure

Reply:
Curious why https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter is no longer a best practice?

Reply:
That is still best practice, and you do want to have getters/setters in order to maintain invariants. This is why I asked to remove it (after the change). Sorry I wasn't exactly clear

Reply:
I don't think it's that simple though, in some cases I see structs with getters and setters for all fields, what's the point of that versus making all fields `pub`?

Reply:
Are you responding to the thread or the commit? I have updated the text.

Reply:
thread